cmd/starbridge/integrations/ethereum.go

```diff
@@ -47,41 +48,60 @@ func Ethereum2Transaction(conn *ethclient.Client, txReceipt *types.Receipt, tx *
 	if !IsMyContractAddress(tx.To().Hex()) {
 		return nil, fmt.Errorf("unsupported receiver address '%s'", tx.To().Hex())
 	}
-
 	fromAddress, e := getFromAddress(tx)
 	if e != nil {
 		return nil, fmt.Errorf("unable to get From address: %s", e)
 	}
 
-	counter, e := Counter.NewCounter(common.HexToAddress(MY_ETHEREUM_CONTRACT_ADDRESS), conn)
-	if e != nil {
-		return nil, fmt.Errorf("unable to construct counter: %s", e)
+	// pulled logic to read events from here: https://goethereumbook.org/event-read/ with many modifications to suit our smart contract
+	// filter log events by a query
+	query := ethereum.FilterQuery{
+		FromBlock: txReceipt.BlockNumber,
+		ToBlock:   txReceipt.BlockNumber,
+		Addresses: []common.Address{
+			common.HexToAddress(MY_ETHEREUM_CONTRACT_ADDRESS),
+		},
 	}
-	cValue, e := counter.GetCount(nil)
+	logs, e := conn.FilterLogs(context.Background(), query)
 	if e != nil {
-		return nil, fmt.Errorf("unable to invoke GetCall(): %s", e)
+		return nil, fmt.Errorf("unable to filter logs for contract address on block number %d", txReceipt.BlockNumber.Int64())
```

Comment:
The error got lost here:
```suggestion
		return nil, fmt.Errorf("unable to filter logs for contract address on block number %d: %w", txReceipt.BlockNumber.Int64(), e)
```

Reply:
will add this in my next PR, thanks!! 🎉